futurecommander_gui/app_web/entry/state.js

```diff
@@ -0,0 +1,100 @@
+
+
+/*
+ * Copyright 2019 François CADEILLAN
+ *
+ * This file is part of FutureCommander.
+ *
+ * FutureCommander is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * FutureCommander is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with FutureCommander.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+const path = require('path');
+
+const { Identity, Functional } = nw.require('openmew-renderer');
+
+const is_entry = ({ resource }) => resource === 'Entry';
```

Comment:
No camelCase ?

Reply:
Nope i prefer to stick with Rust Lint conventions which use snake_case for functions and CamelCase for structures définitions.

If i don't it would be way harder to switch between JS & rust code easily.

Reply:
So everybody will be allowed to use camelCase in Rust parts ? :).

You could use a linter that will help you to think about it: https://eslint.org/docs/rules/camelcase

Reply:
Everybody already *have to* write Rust `struct`,  `enum`, `enum variants`, `type` & `trait` with CamelCase and any `fn` ( including methods ) or `struct` field in snake_case.
There is a rust linter already to ensure that in the CI pipe : clippy.
I'll add ESlint later thus configure it as close as possible to clippy default rules.